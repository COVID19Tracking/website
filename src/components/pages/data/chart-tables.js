import React from 'react'
import { Table, Td, Th } from '~components/common/table'
import { FormatDate } from '~components/utils/format'

// const ChartTables = ({history, usHistory,annotations,testSource,testUnits}) => (

const perMillion = percent => Math.round(percent * 1000000)

const ChartTables = ({ history, usHistory }) => (
  <>
    <h2>New cases</h2>
    <Table>
      <thead>
        <tr>
          <Th header alignLeft>
            Date
          </Th>
          <Th isFirst>State</Th>
          <Th>State 7-day average</Th>
          <Th>State 1/million</Th>
          <Th isFirst>National 1/million</Th>
        </tr>
      </thead>
      <tbody>
        {history.map(row => (
          <tr key={`cases-${row.date}`}>
            <Td alignLeft>
              <FormatDate date={row.date} format="LLL d yyyy" />
            </Td>
            <Td isFirst>{row.positiveIncrease}</Td>
            <Td>Avg</Td>
            <Td>{perMillion(row.childPopulation.positiveIncrease.percent)}</Td>
            <Td>
              {perMillion(
                usHistory.find(item => item.date === row.date).childPopulation
                  .positiveIncrease.percent,
              )}
            </Td>
          </tr>
        ))}
      </tbody>
    </Table>
  </>
)

export default ChartTables
