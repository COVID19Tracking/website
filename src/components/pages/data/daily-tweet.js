/* eslint-disable */
import React from 'react'
import { useStaticQuery, graphql } from 'gatsby'
import Tweet from '~components/common/tweet'

const DailyTweet = () => {
  const data = useStaticQuery(graphql`
    {
      allTweets(
        filter: { is_pinned: { eq: true } }
        sort: { fields: date, order: DESC }
        limit: 1
      ) {
        nodes {
          id_str
          date(formatString: "MMMM D yyyy")
          entities {
            urls {
              display_url
            }
            media {
              media_url
            }
          }
          full_text
        }
      }
    }
  `)

  if (!data || !data.allTweets) {
    return null
  }

  const { id_str, full_text, entities, date } = data.allTweets.nodes[0]

  return (
    <Tweet
      text={full_text}
      media={entities.media[0] && entities.media[0].media_url}
      link={`https://twitter.com/COVID19Tracking/status/${id_str}`}
      date={date}
    />
  )
}

export default DailyTweet
