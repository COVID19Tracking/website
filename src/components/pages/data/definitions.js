import React, { useState } from 'react'
import { Row, Col } from '~components/common/grid'
import definitionStyles from './definitions.module.scss'

const Definitions = ({ definitions, order }) => {
  const orderedDefinitions = []
  order.forEach(field => {
    orderedDefinitions.push(
      definitions.find(definition => definition.fieldName === field),
    )
  })
  const [expanded, setExpanded] = useState([])
  return (
    <Row className={definitionStyles.definitions}>
      <Col width={[4, 6, 3]}>
        <h3 className={definitionStyles.header}>Definitions</h3>
        <button
          className={definitionStyles.expandAll}
          type="button"
          onClick={() => {
            if (expanded.length === orderedDefinitions.length) {
              setExpanded([])
              return
            }
            setExpanded(Array.from(orderedDefinitions.keys()))
          }}
        >
          {expanded.length === orderedDefinitions.length ? (
            <>Collapse all</>
          ) : (
            <>Expand all</>
          )}
        </button>
      </Col>
      <Col width={[4, 6, 9]} paddingLeft={[0, 0, 8]}>
        {orderedDefinitions.map((definition, key) => (
          <div key={`definition-${definition.name}`}>
            <button
              type="button"
              className={definitionStyles.definitionButton}
              aria-expanded={expanded.indexOf(key) > -1}
              aria-controls={`definition-pane--${key}`}
              onClick={event => {
                event.preventDefault()
                const current = [...expanded]
                if (expanded.indexOf(key) > -1) {
                  current.splice(expanded.indexOf(key), 1)
                } else {
                  current.push(key)
                }
                setExpanded(current)
              }}
            >
              {definition.name}{' '}
              <span className={definitionStyles.arrowDown} aria-hidden>
                ↓
              </span>{' '}
              <span className={definitionStyles.arrowUp} aria-hidden>
                ↑
              </span>
            </button>
            <div
              id={`definition-pane--${key}`}
              className={definitionStyles.pane}
              hidden={expanded.indexOf(key) === -1}
              dangerouslySetInnerHTML={{
                __html:
                  definition.childContentfulDataDefinitionDefinitionTextNode
                    .childMarkdownRemark.html,
              }}
            />
          </div>
        ))}
      </Col>
    </Row>
  )
}

export default Definitions
