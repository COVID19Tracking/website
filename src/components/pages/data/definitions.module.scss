.definition-button {
  @include remove-button-style();
  @include margin(16, top);
  display: block;
  font-weight: bold;
  text-align: left;
  .arrow-down,
  .arrow-up {
    @include margin(8, left);
  }
  .arrow-up {
    display: none;
  }
  .arrow-down {
    display: inline-block;
  }
  &[aria-expanded='true'] {
    .arrow-up {
      display: inline-block;
    }
    .arrow-down {
      display: none;
    }
  }
}
