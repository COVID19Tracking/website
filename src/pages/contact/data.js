/* eslint-disable no-unused-vars */
import React, { useState } from 'react'
import { graphql } from 'gatsby'
import Layout from '~components/layout'
import { Form, Textarea, Input, Select, List } from '~components/common/form'
import { Row, Col } from '~components/common/grid'
import Recaptcha from '~components/common/recaptcha'
import Alert from '~components/common/alert'

const ContactPage = ({ data }) => {
  const [name, setName] = useState(false)
  const [message, setMessage] = useState(false)
  const [date, setDate] = useState(false)
  const [email, setEmail] = useState(false)

  const isComplete = () => name && message && date && email

  return (
    <Layout
      title="Contact Us: Data"
      socialCard={{
        description:
          'The COVID Tracking Project runs on the effort and diligence of hundreds of volunteers, and we welcome your contribution.',
      }}
      centered
    >
      <Form
        method="POST"
        name="fa-form-1"
        action="https://webhook.frontapp.com/forms/c3c09a5603c02d2b7f86/KU--_qKvaSkUZHR2T92KQZ8hy5TatxTkRexeSE4NtpFltSVYROQxxbkIdSNH3RcHlZUI4RGGlHL7NwDr00Ki8WMFSOoBBMmgZ28PppSbXx-SQVi51ogAj28bbn6M"
        encType="multipart/form-data"
        acceptCharset="utf-8"
        referrerPolicy="unsafe-url"
      >
        <Select
          isRequired
          label="State"
          detailText="Please let us know which state's data you have a question or issue about"
          id="contact-state"
          name="state"
          options={[
            'All states (US Data)',
            ...data.allCovidStateInfo.nodes.map(node => node.name),
          ]}
        />
        <Input
          type="date"
          label="Date when issue began"
          id="contact-date"
          name="contact-date"
          isRequired
          onChange={event => setDate(event.target.value)}
        />
        <List
          type="checkbox"
          id="contact-data"
          name="data-fields"
          label="Which metric are you writing us about?"
          isRequired
          options={[
            'Cases',
            'Tests',
            'Hospitalization',
            'Recovery',
            'Death',
            'Other',
          ].map(item => ({
            value: item,
            label: item,
          }))}
        />
        <Row>
          <Col width={[4, 6, 6]} paddingRight={[0, 0, 32]}>
            <Input
              isRequired
              label="Your name"
              type="text"
              name="name"
              id="contact-name"
              onChange={event => {
                setName(event.target.value)
              }}
            />
          </Col>
          <Col width={[4, 6, 6]} paddingLeft={[0, 0, 8]}>
            <Input
              isRequired
              type="email"
              label="Your email address"
              name="email"
              id="contact-email"
              onChange={event => setEmail(event.target.value)}
            />
          </Col>
        </Row>
        <Textarea
          label="Please describe the issue"
          name="body"
          id="contact-message"
          isRequired
          onChange={event => setMessage(event.target.value)}
        />

        <input
          type="text"
          aria-hidden
          style={{ display: 'none' }}
          name="subject"
          tabIndex="-1"
          value={`${name} - Data issues`}
        />
        <br />
        <input
          type="hidden"
          name="autoreply-from"
          value="contact@covidtracking.com"
        />
        <input
          type="hidden"
          name="autoreply-sender-name"
          value="The COVID Tracking Project"
        />
        {!isComplete() && (
          <div>
            <Alert>You must complete all fields</Alert>
          </div>
        )}
        <div>
          <Recaptcha />
        </div>
        <button
          type="submit"
          disabled={!isComplete()}
          aria-disabled={!isComplete()}
        >
          Contact us
        </button>
      </Form>
    </Layout>
  )
}

export default ContactPage

export const query = graphql`
  query {
    allCovidStateInfo(sort: { fields: name }) {
      nodes {
        name
      }
    }
  }
`
