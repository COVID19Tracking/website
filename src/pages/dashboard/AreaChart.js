/* eslint-disable no-debugger */
import React, { useState, useEffect } from 'react'
import { json } from 'd3-fetch'
import D3AreaChart from './charts/_AreaChart'
import { sortByDate, transformUsDaily } from './utils'
import { COLOR_MAP } from './constants'

// unless async data is shared across many components,
// each component should make their own api calls and have their own state
const AreaChart = () => {
  const [data, updateData] = useState([])

  useEffect(() => {
    async function fetchUsDailyData() {
      const usDaily = await json('https://covidtracking.com/api/us/daily')
      const sortedUsDaily = sortByDate(usDaily)
      const transformed = transformUsDaily(sortedUsDaily)
      updateData(transformed)
    }
    fetchUsDailyData()
  }, [])

  return (
    <>
      {/* <pre>{JSON.stringify(data, null, 2)}</pre> */}
      <D3AreaChart
        data={data}
        fill={d => (d === 'Total' ? COLOR_MAP.TOTAL : COLOR_MAP.POSITIVE)}
        height={400}
        labelOrder={['Total', 'Positive']}
        marginBottom={40}
        marginLeft={80}
        marginRight={10}
        marginTop={10}
        xTicks={2}
        width={400}
      />
    </>
  )
}
export default AreaChart
