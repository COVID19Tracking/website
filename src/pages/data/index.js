import React from 'react'
import { graphql } from 'gatsby'

import DetailText from '~components/common/detail-text'
import Container from '~components/common/container'
import Layout from '~components/layout'
import ContentfulContent from '~components/common/contentful-content'
import States from '~components/pages/data/states'

import { DownloadDataRow } from '~components/pages/state/download-data'
import Summary from '~components/pages/data/summary'
import SummaryCharts from '~components/pages/data/summary-charts'
import DailyTweet from '~components/pages/data/daily-tweet'

const DataPage = ({ data }) => {
  const stateNavList = []
  data.allCovidStateInfo.nodes.forEach(node => {
    stateNavList.push(node)
  })
  const pageDescription = 'Our most up-to-date data on COVID-19 in the US.'
  return (
    <Layout
      title="Our Data"
      description={pageDescription}
      socialCard={{
        description: pageDescription,
      }}
      path="/data"
    >
      <ContentfulContent
        content={data.dataPreamble.content.childMarkdownRemark.html}
        id={data.dataPreamble.contentful_id}
      />

      <DailyTweet />
      <DownloadDataRow
        slug="all-states"
        lastUpdateEt={data.lastUpdate.nodes[0].date}
        national
      />
      <Summary
        stateSlug="national"
        data={data.covidUs}
        sevenDaysAgo={data.usSevenDaysAgo}
        national
      />
      <Container narrow>
        <DetailText>
          <div
            dangerouslySetInnerHTML={{
              __html:
                data.nationalSummaryFootnote.content.childMarkdownRemark.html,
            }}
          />
        </DetailText>
      </Container>
      <SummaryCharts
        history={data.allCovidUsDaily.nodes}
        annotations={data.allContentfulChartAnnotation}
        national
      />

      <States
        states={data.allCovidStateInfo.nodes}
        stateData={data.allCovidState.nodes}
        sevenDaysAgoList={data.allCovidStateDaily.nodes}
        stateMetadata={data.allContentfulStateOrTerritory.nodes}
      />
    </Layout>
  )
}

export default DataPage

export const query = graphql`
  query($sevenDaysAgo: Date) {
    lastUpdate: allCovidUsDaily(sort: { fields: date, order: DESC }, limit: 1) {
      nodes {
        date(formatString: "MMMM D, YYYY")
      }
    }
    nationalSummaryFootnote: contentfulSnippet(
      slug: { eq: "national-summary-footnote" }
    ) {
      id
      contentful_id
      name
      content {
        childMarkdownRemark {
          html
        }
      }
    }
    dataPreamble: contentfulSnippet(slug: { eq: "data-preamble" }) {
      id
      contentful_id
      name
      content {
        childMarkdownRemark {
          html
        }
      }
    }
    covidUs {
      positive
      positiveIncrease
      negative
      pending
      hospitalizedCurrently
      hospitalizedCumulative
      inIcuCurrently
      inIcuCumulative
      recovered
      onVentilatorCurrently
      onVentilatorCumulative
      death
      totalTestResults
      totalTestResultsIncrease
    }
    usSevenDaysAgo: covidUsDaily(date: { eq: $sevenDaysAgo }) {
      positive
      totalTestResults
    }
    allCovidStateInfo(sort: { fields: name }) {
      nodes {
        name
        state
        notes
        covid19Site
        covid19SiteSecondary
        twitter
        childSlug {
          slug
        }
      }
    }
    allCovidUsDaily {
      nodes {
        date(formatString: "YYYYMMDD")
        totalTestResultsIncrease
        positiveIncrease
        hospitalizedCurrently
        deathIncrease
        childPopulation {
          deathIncrease {
            percent
          }
          hospitalizedCurrently {
            percent
          }
          positiveIncrease {
            percent
          }
          totalTestResultsIncrease {
            percent
          }
        }
      }
    }
    allCovidState {
      nodes {
        dataQualityGrade
        death
        deathConfirmed
        deathProbable
        hospitalizedCumulative
        hospitalizedCurrently
        inIcuCumulative
        inIcuCurrently
        lastUpdateEt
        dateModified(formatString: "MMM D, YYYY h:mm a")
        negative
        negativeTestsViral
        onVentilatorCumulative
        onVentilatorCurrently
        pending
        positive
        positiveCasesViral
        probableCases
        positiveIncrease
        positiveTestsViral
        posNeg
        recovered
        state
        totalTestResults
        totalTestResults
        totalTestsPeopleViral
        totalTestsViral
        totalTestEncountersViral
        totalTestsAntibody
      }
    }
    allCovidStateDaily(filter: { date: { eq: $sevenDaysAgo } }) {
      nodes {
        positive
        state
      }
    }
    allContentfulStateOrTerritory {
      nodes {
        testUnitsUnknown
        code
      }
    }
    allContentfulChartAnnotation(
      filter: { displayNationalChart: { eq: true } }
      sort: { fields: date, order: DESC }
    ) {
      nodes {
        title
        description {
          description
        }
        date(formatString: "YYYYMMDD")
        dataElement
        contentful_id
        childContentfulChartAnnotationDescriptionTextNode {
          childMarkdownRemark {
            html
          }
        }
      }
    }
  }
`
