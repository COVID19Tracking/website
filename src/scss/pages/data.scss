@import '../colors.scss';

.data-state {
  .state-notes {
    font-size: 0.8rem;
  }

  p {
    margin-bottom: 0;
  }
}

.top-link {
  color: $color-grey-4;
  margin-top: 1.8rem;
  display: inline-block;
}

li {
	margin-bottom: 0;
}

.state-nav {
  h3 {
    margin-bottom: 0.5rem;
  }

  li {
    display: inline;
    padding: 0.25rem 0.75rem;
  }

  ul {
    list-style: none;
    margin: 0;
    display: flex;
    flex-wrap: wrap;
    justify-content: space-evenly;
  }
}
