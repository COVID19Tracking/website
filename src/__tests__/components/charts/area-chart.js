import React from 'react'
import renderer from 'react-test-renderer'
import data from 'sample-chart-data'
import AreaChart from '../../../components/charts/area-chart'

describe('Components : Charts: Area chart', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <AreaChart
          fill={d => (d === 'Group A' ? '#333' : '#000')}
          data={data}
          xTicks={2}
          height={200}
          width={200}
          marginBottom={30}
          marginTop={30}
          marginLeft={30}
          marginRight={30}
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
