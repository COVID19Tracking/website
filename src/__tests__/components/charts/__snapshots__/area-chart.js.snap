// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components : Charts: Area chart renders correctly 1`] = `
<svg
  className="chart"
  onMouseLeave={[Function]}
  onMouseMoveCapture={[Function]}
  onTouchEndCapture={[Function]}
  onTouchStart={[Function]}
  viewBox="0 0 200 200"
>
  <g
    transform="translate(30 30)"
  >
    <g
      transform="translate(0 140)"
    >
      <text
        className="label xTickLabel"
        x={29.166666666666668}
        y={20}
      >
        Mar  4
      </text>
      <text
        className="label xTickLabel"
        x={99.16666666666667}
        y={20}
      >
        Mar  4
      </text>
    </g>
    <g>
      <g>
        <svg
          className="yTickLabel"
          x="-10"
          y={144}
        >
          <text
            className="label"
          >
            0
          </text>
        </svg>
        <line
          className="gridLine"
          x1={0}
          x2={140}
          y1={140}
          y2={140}
        />
      </g>
      <g>
        <svg
          className="yTickLabel"
          x="-10"
          y={121.30956239870339}
        >
          <text
            className="label"
          >
            200
          </text>
        </svg>
        <line
          className="gridLine"
          x1={0}
          x2={140}
          y1={117.30956239870339}
          y2={117.30956239870339}
        />
      </g>
      <g>
        <svg
          className="yTickLabel"
          x="-10"
          y={98.61912479740681}
        >
          <text
            className="label"
          >
            400
          </text>
        </svg>
        <line
          className="gridLine"
          x1={0}
          x2={140}
          y1={94.61912479740681}
          y2={94.61912479740681}
        />
      </g>
      <g>
        <svg
          className="yTickLabel"
          x="-10"
          y={75.92868719611022}
        >
          <text
            className="label"
          >
            600
          </text>
        </svg>
        <line
          className="gridLine"
          x1={0}
          x2={140}
          y1={71.92868719611022}
          y2={71.92868719611022}
        />
      </g>
      <g>
        <svg
          className="yTickLabel"
          x="-10"
          y={53.23824959481361}
        >
          <text
            className="label"
          >
            800
          </text>
        </svg>
        <line
          className="gridLine"
          x1={0}
          x2={140}
          y1={49.23824959481361}
          y2={49.23824959481361}
        />
      </g>
      <g>
        <svg
          className="yTickLabel"
          x="-10"
          y={30.547811993517023}
        >
          <text
            className="label"
          >
            1,000
          </text>
        </svg>
        <line
          className="gridLine"
          x1={0}
          x2={140}
          y1={26.547811993517023}
          y2={26.547811993517023}
        />
      </g>
      <g>
        <svg
          className="yTickLabel"
          x="-10"
          y={7.857374392220418}
        >
          <text
            className="label"
          >
            1,200
          </text>
        </svg>
        <line
          className="gridLine"
          x1={0}
          x2={140}
          y1={3.8573743922204184}
          y2={3.8573743922204184}
        />
      </g>
    </g>
  </g>
  <g
    transform="translate(30 30)"
  >
    <path
      d="M0,140L140,140L140,126.0453808752026L0,0Z"
      fill="#333"
      opacity={0.8}
    />
    <path
      d="M0,140L140,140L140,125.02431118314425L0,0.11345218800648427Z"
      fill="#000"
      opacity={0.8}
    />
  </g>
</svg>
`;
